### PR TITLE
test images: Image Promoter fixes

### PR DESCRIPTION
**Pull request by claudiubelu**
**What type of PR is this?**

/kind cleanup
/sig testing
/area conformance

**What this PR does / why we need it**:

Prior to the Image Centralization part 4 (https://github.com/kubernetes/kubernetes/pull/81170), a PR merged that enables the Image Promoter to run on the k/k test images.

The Image Promoter currently only builds the Conformance-related images, but the Image Centralization part 4 centralized some of those images into ``agnhost``, so they need to be removed from the ``conformance_images`` list.

Additionally, https://github.com/kubernetes/kubernetes/pull/81226 proposes ``mounttest-user`` image to be removed, and ``RunAsUser`` to be used in tests instead. This was proposed here: https://github.com/kubernetes/kubernetes/issues/76342

The image used by the Image Promoter (``gcr.io/k8s-testimages/gcb-docker-gcloud:v20190906-745fed4``) is based on busybox, and thus, the sed binary is actually busybox. ``image-util.sh`` calls ``kube::util::ensure-gnu-sed`` several times, which ensures that a GNU sed binary exists (it checks by greping GNU in its --help output). Obviously, it won't match the busybox sed binary. But the sed usage in ``image-util.sh`` is fairly simple, and the busybox sed is sufficient.

Bumps image versions for: ``jessie-dnsutils``, ``nonewprivs``, ``resource-consumer``, ``sample-apiserver``. These images are included in the ``conformance_images`` that are being built by the Image Promoter, so we're bumping them just to make sure we're not breaking anything and cause all the CIs to fall. We're going to bump the image versions used in tests in a subsequent PR. The image version was not bumped for: ``agnhost``, ``kitten``, ``nautilus``, as they were already bumped by the Image Centralization part 4 PR.

**Which issue(s) this PR fixes**:

Related: #76342
Related: #81226
Related: #81170
Related: #87251

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: hack/lib/util.sh [hunk 1]
```diff
@@ -698,7 +698,9 @@ function kube::util::ensure_dockerized {
 #  SED: The name of the gnu-sed binary
 #
 function kube::util::ensure-gnu-sed {
-  if LANG=C sed --help 2>&1 | grep -q GNU; then
+  # NOTE: the echo below is a workaround to ensure sed is executed before the grep.
+  # see: https://github.com/kubernetes/kubernetes/issues/87251
+  if echo "$(LANG=C sed --help 2>&1)" | grep -q "GNU\|BusyBox"; then
```
Comment (listx) [inline hack/lib/util.sh]: I strongly doubt that the shell runs the piped commands randomly run out-of-order. The stdout logs that we've seen is probably a result of the buffered output being re-combined incorrectly.

Comment (claudiubelu) [reply hack/lib/util.sh]: Indeed, it sounds incredible, but it is also the only explanation as to why that function would randonly fail.

Comment (listx) [reply hack/lib/util.sh]: Are we sure that this function fails randomly *on the same system*, and not that the function is failing (seemingly) randomly on different systems that have (or don't have) GNU sed?

Comment (claudiubelu) [reply hack/lib/util.sh]: This is basically what is being executed: https://paste.ubuntu.com/p/2jHTmfCvpP/

In my ubuntu 16.04 (GNU bash, version 5.0.0(1)-release (x86_64-pc-linux-gnu) and my 18.04 (GNU bash, version 4.4.20(1)-release (x86_64-pc-linux-gnu) OSes, it passed for all 200 runs.

While for the ``gcr.io/k8s-testimages/gcb-docker-gcloud:v20190906-745fed4`` image ("Alpine Linux v3.10") (GNU bash, version 5.0.0(1)-release (x86_64-alpine-linux-musl), the number of failures varies between 19 and 98.

**Comment by spiffxp** [comment]
/retest
flake https://github.com/kubernetes/kubernetes/issues/87441

**Pull request merged**